data_juicer/ops/mapper/video_audio_speech_ASR_mapper.py

```diff
@@ -0,0 +1,105 @@
+        super().__init__(*args, **kwargs)
+        self._batched_op = True
+        self._model_sampling_rate = 16000
+        self.model_dir_ASR = model_dir_ASR
+
+        self.model_key = prepare_model(
+            model_type='SenseVoiceSmall',
+            pretrained_model_name_or_path=model_dir_ASR,
+        )
+
+        self.speech_ASR = speech_ASR
+
+    def process_single(self, sample, rank=None):
+        # check if it's generated already
+        if MetaKeys.speech_emotion in sample[Fields.meta]:
+            return sample
+
+        # there is no video in this sample
+        if self.video_key not in sample or not sample[self.video_key]:
+            sample[Fields.source_file] = []
+            return sample
+        
+        if not MetaKeys.video_audio_tags in sample[Fields.meta]:
+            raise ValueError("video_active_speaker_mapper must be operated after video_tagging_from_audio_mapper.")
+
+
+        # load video paths
+        loaded_video_keys = sample[self.video_key]
+        audio_tags = sample[Fields.meta][MetaKeys.video_audio_tags]
+
+        ASR_model, kwargs1= get_model(self.model_key, rank=rank)
+
+        # model, feature_extractor = get_model(self.model_key, rank=rank)
+        video_audio_tags = []
+
+        for id,video_path in enumerate(loaded_video_keys):
+            if audio_tags[id] == 'Speech':
+                # only extract audio data and sr for index 0 for now
+                ys, srs, valid_indexes = extract_audio_from_video(
+                    video_path, stream_indexes=[0])
+                if len(valid_indexes) == 0:
+                    # there is no valid audio streams. Skip!
+                    video_audio_tags.append(self._no_audio_label)
+                    continue
+
+                # inference
+                y = ys[0]
+                sr = srs[0]
+                # check if it meets the sampling rate condition of the model
+                if sr != self._model_sampling_rate:
+                    y = librosa.resample(y,
+                                        orig_sr=sr,
+                                        target_sr=self._model_sampling_rate)
+                    sr = self._model_sampling_rate
+                
+                inputs = torch.tensor(y).to(next(ASR_model.parameters()).device)
+                with torch.no_grad():
+                    output_ASR_emo = ASR_model.inference(
+                        data_in=inputs,
+                        language="auto", # "zn", "en", "yue", "ja", "ko", "nospeech"
+                        use_itn=False,
+                        **kwargs1,
+                    )
+                
+                video_audio_tags.append({'language':output_ASR_emo[0][0]['text'].split('<|',1)[-1].split('|>')[0], 'asr': output_ASR_emo[0][0]['text'].split('|>',4)[-1]})
+            else:
+                video_audio_tags.append('')
+            
+        sample[Fields.meta][self.speech_ASR] = video_audio_tags
+        gc.collect()
```

Comment:
no need to do sample-wise gc and empty_cache, maybe you can just modify the base class in batch_process func, or leave a "TODO remark", and later I and @HYLcool will try to fix this.